include/lemon/parallel.hpp

```diff
@@ -1,7 +1,10 @@
 #ifndef LEMON_PARALLEL_HPP
 #define LEMON_PARALLEL_HPP
 
+#include <list>
```

Comment:
why use a linked list instead of a vector? There is usually no advantage to linked-list

Reply:
The original idea was that a list has constant insertion time. Benchmarking showed a list was slightly better (within the margin of error though).

Reply:
OK, interesting. Even with pre-allocation of the vector? list usually have a better theoretical algorithmic performance, but worst in-practise, since they fragment memory a lot.

Reply:
I never tried pre-allocation!